### PR TITLE
sort by alphabetical order for languages

### LINKED ISSUE #6057: More Filters: UI: Sorting of filters: required for user testing
The ordering of the filtering

At the moment the API returns this based on count for subjects and genres.

Suggested change is to order this on the front end for now in alphabetical order.

We discussed that we will do this only for languages and then test this.

### PR DESCRIPTION
**Pull request by ishouty**
Ability to sort aggregation filter on the front end in alphabetical or count.

closes #6057

### CODE REVIEW

File: common/utils/filters.ts [hunk 1]
```diff
@@ -115,3 +128,21 @@ export const getSelectedFilterColor = (form: HTMLFormElement): string => {
       : imagesColorValue;
   return imagesColor;
 };
+
+export const sortAggregationBucketByOrder = (
```
Comment (gestchild) [inline common/utils/filters.ts:132]: not a big deal, but could this just be `sortAggregationBucket`

**Review by gestchild — APPROVED** [review]

File: common/utils/filters.ts [hunk 2]
```diff
@@ -115,3 +128,21 @@ export const getSelectedFilterColor = (form: HTMLFormElement): string => {
...
+      sortedfilter = filter.sort(sortByCount);
+      break;
+    case 'alphabetical':
+      sortedfilter = filter.sort(sortLabelByAlphabeticalOrder);
+      break;
+    default:
+      sortedfilter = filter;
+  }
+  return sortedfilter;
+};
```
Comment (jamesgorrie) [inline common/utils/filters.ts:148]: You can just return the value you need in the case statement without introducing variables.

```ts
switch (sortBy) {
    case 'count':
      return filter.sort(sortByCount);
      break;
    case 'alphabetical':
      return filter.sort(sortLabelByAlphabeticalOrder);
      break;
    default:
      return filter;
  }
```

File: common/utils/filters.ts [hunk 3]
```diff
@@ -115,3 +128,21 @@ export const getSelectedFilterColor = (form: HTMLFormElement): string => {
       : imagesColorValue;
   return imagesColor;
 };
+
+export const sortAggregationBucket = (
+  filter: CatalogueAggregationBucket[],
+  sortBy?: 'alphabetical' | 'count'
```
Comment (jamesgorrie) [inline common/utils/filters.ts:134]: `sortBy: 'alphabetical' | 'count' = 'count'`

It seems weird to offer a method that has the potential to do nothing.

Comment (ishouty) [reply common/utils/filters.ts:134]: not all the aggregations atm return by count, which i thought we could use. Happy to remove this..

**Pull request merged**